### DISCUSSION TITLE
Passing model array to child very slow, adds 100+ms on very simple model  with query time ~170μs

### DISCUSSION THREAD

**Discussion by rstanislav** [livewire/livewire#3612] [7. Bugs (archived)]
Hello!

I have very simple model:

```
$table->engine = 'InnoDB';
$table->increments('id');
$table->string('name')->nullable();
$table->string('description')->nullable();
$table->softDeletes();
$table->timestamps();
```

With 150 rows in it, description field is empty, only name used (3-20 characters).

In Livewire controller i have:
```
public $manufacturers = null;

// in mount function
$this->manufacturers = CarManufacturer::orderBy('name')->get();
```

Then in blade i include another Livewire element passing manufacturers along other few elements :

```
@foreach ($client_cars as $client_car)
                    @livewire('one-client-car', ['client_car' => $client_car, 'manufacturers' => $manufacturers,
                    'models' => $models],
                    key($client_car->id))
@endforeach
```

Inside one-client-car i have:

```
public function mount($client_car, $manufacturers, $models)
    {
        $this->manufacturers = $manufacturers;
        $this->models = $models;
    }

```
And then i display that array in foreach loop inside blade file:

```
<select wire:model.lazy="manufacturer_id"
                        class="block w-full px-3 py-2 mt-1 border border-gray-300 rounded-md shadow-sm">
                        <option selected="selected">{{ $manufacturer_name }}</option>
                        @foreach ($manufacturers as $manufact)
                            <option value="{{ $manufact->id }}">
                                {{ $manufact->name }}
                            </option>
                        @endforeach
                    </select>
```
And at page load(with 4 child elements in foreach loop) i see over 250 ms request time. Even if model in child element not used. If i don't load model and don't pass it to child elements req time is ~130ms (opcache turned off and dev mode on) - tested with opcache on - got about 200ms req time vs normal 25-35ms. 

Strange thing is - according to debugbar request to fetch manufacturers is 170μs:

```
select `name`, `id` from `car_manufacturers` where `car_manufacturers`.`deleted_at` is null order by `name` asc
170μs
```

If i change logic and do:
`
$this->manufacturers = CarManufacturer::orderBy('name')->pluck('name', 'id')->toArray();`

And then display it with

```
<select wire:model.lazy="manufacturer_id"
                        class="block w-full px-3 py-2 mt-1 border border-gray-300 rounded-md shadow-sm">
                        <option selected="selected">{{ $manufacturer_name }}</option>
                        @foreach ($manufacturers as $manufacturers_id => $manufacturers_name)
                            <option value="{{ $manufacturers_id }}">
                                {{ $manufacturers_name }}
                            </option>
                        @endforeach
                    </select>
```
There is no additional req time - it always stays around 130 without opcache and with opcache around 25, even with 1500 rows in manufacturer table.

Very strange is that Queries tab of debugbar doesnot display any increase in req time - have about 16 queries for full page, and avr time is around 120us for each, don't have even single 1ms query, only timeline shows that "Application" time increases to 200+ms.

If i add second model ("models" in code above) it adds another 100ms

It is def not DB prob because query time the same (with pluck or without).

Help me please, how to trace this issue ?

Laravel: latest ver
Livewire: latest ver

**Comment by ariaieboy** [comment]
you can use tracing tools to find the bottleneck of your script.
one of them is [tracing-laravel](https://github.com/Vinelab/tracing-laravel)

trace your application and find the bottleneck and then we can work on that.
